runtime/server/server.go

```diff
@@ -82,11 +83,18 @@ func (s *Server) ServeHTTP(ctx context.Context) error {
 
 // ErrorToCode returns the Code of the error if it is a Status error
 // otherwise use status.FromContextError to determine the Code.
-// Log level for error codes is defined in logging.DefaultServerCodeToLevel
+// Log level for error codes is defined in logging.DefaultServerCodeToLevel.
 func ErrorToCode(err error) codes.Code {
-	if se, ok := err.(interface {
-		GRPCStatus() *status.Status
-	}); ok {
+	// Commenting this need to test it if works fine or not
+	// if se, ok := err.(interface {
+	// 	GRPCStatus() *status.Status
+	// }); ok {
+	// 	return se.GRPCStatus().Code()
+	// }
+
+	var se interface{ GRPCStatus() *status.Status }
+	ok := errors.As(err, &se)
+	if ok {
```

Comment:
Both the previous and new version here looks weird. Can we find an actual type to cast as? Look in grpc gateway docs

Reply:
As per this [ref](https://github.com/grpc/grpc-go/blob/v1.51.0/status/status.go#L101) i can see we have cast as GRPCStatus() *status.Status only, any other doc i should refer?

Reply:
Actually, I think I removed this in main today, so after merge, this probably won't be a problem